src/frontend/lexer.rs

```diff
@@ -0,0 +1,259 @@
+
+        let next_token = match next_char {
+            '(' => self.accept(LParen),
+            ')' => self.accept(RParen),
+            '\'' => self.accept(Quote),
+            c if Self::starts_identifier(c) => self.parse_identifier(),
+            c if Self::starts_integer(c) => self.parse_integer(),
+            c if Self::starts_string(c) => self.parse_string(),
+            // Unexpected symbol
+            c => Err(Error::UnexpectedSymbol(c.clone())),
+        };
+
+        Some(next_token)
+    }
+
+    // ---------------------------------------------------------------------- //
+    // Whitespace handling                                                    //
+    // ---------------------------------------------------------------------- //
+
+    fn is_whitespace(c: &char) -> bool {
+        c.is_ascii_whitespace()
+    }
+    fn skip_whitespace(&mut self) {
+        self.skip_while(Self::is_whitespace);
+    }
+
+    // ---------------------------------------------------------------------- //
+    // Token specific lexing methods                                          //
+    // ---------------------------------------------------------------------- //
+
+    // Identifier ----------------------------------------------------------- //
+    fn starts_identifier(c: &char) -> bool {
+        c.is_ascii_graphic() && !c.is_ascii_digit() && !"()'\"".contains(*c)
+    }
+    fn parse_identifier(&mut self) -> TokResult {
+        let matcher = |c: &char| c.is_ascii_graphic() && !"()\"".contains(*c);
+        let string = self.collect_while(matcher);
+
+        Ok(match &string[..] {
+            "true"  => Token::Boolean(true),
+            "false" => Token::Boolean(false),
+            _ => Token::Identifier(string),
+        })
+    }
+
+    // Integer -------------------------------------------------------------- //
+    fn starts_integer(c: &char) -> bool {
+        c.is_ascii_digit()
+    }
+    fn parse_integer(&mut self) -> TokResult {
+        Ok(Token::Integer(self.collect_while(char::is_ascii_digit)))
+    }
+
+    // String --------------------------------------------------------------- //
+    fn starts_string(c: &char) -> bool {
+        *c == '"'
+    }
+    fn parse_string(&mut self) -> TokResult {
+        self.consume();
+        let contents = self.collect_while(|&c| c != '"');
+
+        if self.peek().is_none() {
+            // Missing the closing " on a string. Reached EOF
+            Err(Error::UnclosedString)
+        } else {
+            self.consume();
+            Ok(Token::String(contents))
+        }
+    }
+
+    // ---------------------------------------------------------------------- //
+    // Helper methods                                                         //
+    // ---------------------------------------------------------------------- //
+
+    // Single element ------------------------------------------------------- //
+    fn peek(&mut self) -> Option<&char> {
```

Comment:
Why arent some methods marked public vs. others private. It would then be clearer which are the helper methods.

Reply:
There isn't a `private` keyword in Rust if I'm not mistaken. So... There isn't much I can do here :)